lib/context_formatter.rb

```diff
@@ -0,0 +1,57 @@
+module Poesie
+  module ContextFormatter
+
+    # Write the JSON output file containing all context keys
+    #
+    # @param [Array<Hash<String, Any>>] terms
+    #        JSON returned by the POEditor API
+    # @param [String] file
+    #        The path of the file to write
+    #
+    def self.write_context_json(terms, file, currentOS)
+
+      json_hash = { "date" => "#{Time.now}" }
+
+      stats = { :filteredOS => 0, :nil => 0, :count => 0 }
+
+      #switch on term / context
+      array_context = Array.new
+      terms.each do |term|
+        (term, definition, comment, context) = ['term', 'definition', 'comment', 'context'].map { |k| term[k] }
+
+        # Filter terms and update stats
+        next if (term.nil? || term.empty? || context.nil? || context.empty?) && stats[:nil] += 1
+        if currentOS == CurrentOS::ANDROID
+          next if (term =~ /_ios$/) && stats[:filteredOS] += 1 # Remove android-specific strings
+        elsif currentOS == CurrentOS::IOS
+          next if (term =~ /_android$/) && stats[:filteredOS] += 1 # Remove android-specific strings
+        end
```

Comment:
This way, if we pass the `XXX_FILTER_REGEX` constant as a parameter named `filter_regex`, then here we can simply use:
```
next if (term =~ filter_regex) && stats[:filteredOS] += 1
```
And no `if`/`elsif` necessary anymore.